Applications/PixelPaint/FilterParams.h

```diff
@@ -0,0 +1,191 @@
+        set_title(builder.string_view());
+
+        resize(200, 250);
+        auto& main_widget = set_main_widget<GUI::Frame>();
+        main_widget.set_frame_shape(Gfx::FrameShape::Container);
+        main_widget.set_frame_shadow(Gfx::FrameShadow::Raised);
+        main_widget.set_fill_with_background_color(true);
+        auto& layout = main_widget.template set_layout<GUI::VerticalBoxLayout>();
+        layout.set_margins({ 4, 4, 4, 4 });
+
+        size_t index = 0;
+        size_t columns = N;
+        size_t rows = N;
+
+        for (size_t row = 0; row < rows; ++row) {
+            auto& horizontal_container = main_widget.template add<GUI::Widget>();
+            horizontal_container.template set_layout<GUI::HorizontalBoxLayout>();
+            for (size_t column = 0; column < columns; ++column) {
+                if (index < columns * rows) {
+                    auto& textbox = horizontal_container.template add<GUI::TextBox>();
+                    textbox.set_preferred_size({ 30, 50 });
+                    textbox.on_change = [&, row = row, column = column] {
+                        auto& element = m_matrix.elements()[row][column];
+                        char* endptr = nullptr;
+                        auto value = strtof(textbox.text().characters(), &endptr);
+                        if (endptr != nullptr)
+                            element = value;
+                        else
+                            textbox.set_text("");
+                    };
+                } else {
+                    horizontal_container.template add<GUI::Widget>();
+                }
+            }
+        }
+
+        auto& norm_checkbox = main_widget.template add<GUI::CheckBox>("Normalize");
+        norm_checkbox.set_checked(false);
+
+        auto& wrap_checkbox = main_widget.template add<GUI::CheckBox>("Wrap");
+        wrap_checkbox.set_checked(m_should_wrap);
+
+        auto& button = main_widget.template add<GUI::Button>("Done");
+        button.on_click = [&](auto) {
+            m_should_wrap = wrap_checkbox.is_checked();
+            if (norm_checkbox.is_checked())
+                normalize(m_matrix);
+            done(ExecOK);
+        };
+    }
+
+    Matrix<N, float> m_matrix {};
+    bool m_should_wrap { false };
+};
+
+template<size_t N>
+struct FilterParameters<Gfx::SpatialGaussianBlurFilter<N>> {
+    static OwnPtr<typename Gfx::SpatialGaussianBlurFilter<N>::Parameters> get()
+    {
+        Matrix<N, float> kernel;
+        auto sigma = 1.0f;
+        auto s = 2.0f * sigma * sigma;
+
+        for (auto x = -(ssize_t)N / 2; x <= (ssize_t)N / 2; x++) {
+            for (auto y = -(ssize_t)N / 2; y <= (ssize_t)N / 2; y++) {
+                auto r = sqrt(x * x + y * y);
+                kernel.elements()[x + 2][y + 2] = (exp(-(r * r) / s)) / (M_PI * s);
+            }
+        }
+
+        normalize(kernel);
+
+        return make<typename Gfx::GenericConvolutionFilter<N>::Parameters>(kernel);
+    }
+};
```

Comment:
This generates a gaussian kernel.

Not sure what sort of distortion you're talking about, but a box blur shouldn't cause any particular distortions (maybe some darkening around the edges if it's set to not wrap?)

Reply:
At `3` colors get distorted in a weird way, at `9` it becomes really dark. Can't remember what it did at `7`. Haven't tried other values.

Reply:
I'll pull this PR and take a looksie.

Reply:
So I copied this from PixelPaint into the `calc_matrix` function in `Compositor.cpp` and it crashes at startup trying to access `V0xffffffff`
```c++
    static Matrix<N, float> calc_matrix()
    {
        Matrix<N, float> kernel;
        auto sigma = 1.0f;
        auto s = 2.0f * sigma * sigma;

        for (auto x = -(ssize_t)N / 2; x <= (ssize_t)N / 2; x++) {
            for (auto y = -(ssize_t)N / 2; y <= (ssize_t)N / 2; y++) {
                auto r = sqrt(x * x + y * y);
                kernel.elements()[x + 2][y + 2] = (exp(-(r * r) / s)) / (M_PI * s);
            }
        }

        Gfx::normalize(kernel);
        return kernel;
    }
```

Reply:
That code crashes with `N == 7`

Reply:
:thinking: weird...
the 5x5 gaussian filter crashes in pixelpaint too, but I assume it generates this ok before crashing in apply()

Reply:
> That code crashes with N == 7

That makes too much sense, `-N/2` would be -3, so it's accessing elements at `-1,n`

Reply:
I'm not quite sure how I ever thought this convolution filter was _correct_.

Reply:
I pushed a fix for this (and the distortions) in a [PR-PR](https://github.com/tomuta/serenity/pull/1) (\*giggle\*)

Reply:
Awesome, thanks!